src/steamship/app/lambda_handler.py

```diff
@@ -159,9 +151,55 @@ def handler(event: Dict, context: Dict = None) -> dict:
             # The below should make it so calls to logging.info etc are also routed to the remote logger
             logging.root.addHandler(logging_handler)
 
-        response = _handler(event, context)
+        try:
+            client = Steamship(
+                **{to_snake_case(k): v for k, v in event.get("clientConfig", {}).items()}
+            )
+        except SteamshipError as se:
+            logging.exception(se)
+            return Response.from_obj(se).dict(by_alias=True)
+        except Exception as ex:
+            logging.exception(ex)
+            return Response.error(
+                code=HTTPStatus.INTERNAL_SERVER_ERROR,
+                message="Plugin/App handler was unable to create Steamship client.",
+                exception=ex,
+            ).dict(by_alias=True)
+
+        response = _handler(event, client, context)
+
+        result = response.dict(by_alias=True)
+        # When created with data > 4MB, data is uploaded to a bucket.
+        # This is a very ugly way to get the deep size of this object
+        data_size = sys.getsizeof(json.dumps(result.get("data", {})))
```

Comment:
```suggestion
        data = json.dumps(result.get("data", None)).encode("UTF-8")
        data_size = sys.getsizeof(data)
```

1. Cache the json.dumps result for upload
2. Get the size of the utf-8 encoded version, as its size is different from the non-encoded version.